src/test/java/org/tarantool/jdbc/JdbcStatementIT.java

```diff
@@ -246,4 +248,35 @@ void testStatementConnection() throws SQLException {
         Statement statement = conn.createStatement();
         assertEquals(conn, statement.getConnection());
     }
+
+    @Test
+    void testCloseOnCompletion() throws SQLException {
+        assertFalse(stmt.isCloseOnCompletion());
+        stmt.closeOnCompletion();
+        assertTrue(stmt.isCloseOnCompletion());
+    }
+
+    @Test
+    void testCloseOnCompletionDisabled() throws SQLException {
+        ResultSet resultSet = stmt.executeQuery("SELECT val FROM test WHERE id=1");
+        assertFalse(stmt.isClosed());
+        assertFalse(resultSet.isClosed());
+
+        resultSet.close();
+        assertTrue(resultSet.isClosed());
+        assertFalse(stmt.isClosed());
+    }
+
+    @Test
+    void testCloseOnCompletionEnabled() throws SQLException {
+        stmt.closeOnCompletion();
+        ResultSet resultSet = stmt.executeQuery("SELECT val FROM test WHERE id=1");
```

Comment:
Just curious: I guess we'll have the following behaviour?

```java
ResultSet rs1 = stmt.executeQuery("...");
stmt.closeOnCompletion();
ResultSet rs2 = stmt.executeQuery("...");
rs2.close()
assertFalse(stmt.isClosed());
```

It is because of discarding a last result when a new one is acquired. However whether it is the fully correct behaviour?

Reply:
Hm... Unfortunately, API doesn't enlighten about that case. For instance, PostgreSQL driver doesn't support it (is broken?) but MySQL driver does.

It looks reasonable excluding `assertFalse(stmt.isClosed());` which should be `assertTrue(stmt.isClosed());` from your example. 
I'd say it can be generalized as `Statement.closeOnCompletion()` works only when `ResultSet.close()` is called explicitly by application code.

So the following cases should be possible:

```java
stmt.closeOnCompletion();
ResultSet rs1 = stmt.executeQuery("...");
ResultSet rs2 = stmt.executeQuery("..."); // this closes rs1 internally but stmt is still open
```

Next example is the same as previous:

```java
ResultSet rs1 = stmt.executeQuery("...");
stmt.closeOnCompletion();
ResultSet rs2 = stmt.executeQuery("..."); // this closes rs1 implicitly but stmt is still open
```

Update executions do not affect the statement even they close last result set if any:

```java
stmt.closeOnCompletion();
ResultSet rs1 = stmt.executeQuery("...");
stmt.executeUpdate("..."); // this closes rs1 implicitly but stmt is still open
ResultSet rs2 = stmt.executeQuery("...");
stmt.executeUpdate("..."); // this closes rs2 implicitly but stmt is still open
ResultSet rs3 = stmt.executeQuery("...");
rs3.close(); // app explictly closes rs3; stmt is implicitly closed
```

Reply:
I found all this stuff unobvious, so at least we should document the current behaviour.

Reply:
Okay, to sum up:
1. A result set (RS) can be closed _internally_ or _externally_;
2. a statement closes the last RS _internally_ each time execute* method is invoked;
3. an app closes RS _externally_ via `ResultSet.close()` method.
4. `closeOnCompletion` takes effect when RS is closed _explicitly_.

I added a short java-doc for `closeOnCompletion`.
I will add more detailed (with examples) docs in scope of #60.

Reply:
Okay.